# Import Library

Menginstal library dan modul yang akan digunakan

In [1]:
!pip install -q split-folders

Menghubungkan Google Drive dengan Google Colab

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Import Data from Drive

Mengakses dataset dari Google Drive

In [6]:
dataset = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset'

Melakukan pembagian data dengan ratio 90% (train) dan 10% (test)

In [7]:
#impor package split-folders
import splitfolders

#Membagi data
splitfolders.ratio(dataset,
                   output='/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data',
                   seed=1337,
                   ratio=(.9, .1),
                   group_prefix=None, move=False)

Copying files: 767 files [00:18, 41.42 files/s]


# Split Data

In [8]:
import os

base_dir = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data'

train_dir = os.path.join(base_dir, 'train')
os.chdir(base_dir)
os.rename("val","test")
test_dir = os.path.join(base_dir, 'test')

train_benign = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/train/benign'
train_malignant = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/train/malignant'
train_normal = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/train/normal'

test_benign = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/test/benign'
test_malignant = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/test/malignant'
test_normal = '/content/drive/MyDrive/Dataset/Breast Cancer Dataset/data/test/normal'

print('total citra train:', len(os.listdir(train_benign))+
                            len(os.listdir(train_malignant))+
                            len(os.listdir(train_normal))
                            )
print('total citra test:', len(os.listdir(test_benign))+
                           len(os.listdir(test_malignant))+
                           len(os.listdir(test_normal)))

total citra train: 689
total citra test: 78


Hasil pembagian data citra yang digunakan untuk train sebanyak 689 citra dan citra yang digunakan untuk test sebanyak 78 citra. Data ini digunakan untuk analisis data Imbalance